File: Serene/Serene.Script/Northwind/Customer/CustomerOrderDialog.cs

namespace Serene.Northwind
{
    using Serenity;
    using System.Runtime.CompilerServices;

    [Imported]
    public class CustomerOrderDialog : OrderDialog
    {
        protected override void UpdateInterface()
        {
            base.UpdateInterface();

            EditorUtils.SetReadOnly(form.CustomerID, true);
        }
    }
}

File: Serene/Serene.Web/Modules/Northwind/Customer/CustomerOrderDialog.ts
/// <reference path="../Order/OrderDialog.ts" />

namespace Serene.Northwind {

    @Serenity.Decorators.registerClass()
    export class CustomerOrderDialog extends OrderDialog {

        constructor() {
            super();
        }

        updateInterface() {
            super.updateInterface();

            Serenity.EditorUtils.setReadOnly(this.form.CustomerID, true);
        }
    }
}